src/hpr_config.erl

```diff
@@ -0,0 +1,178 @@
+-module(hpr_config).
+
+-export([
+    init/0,
+    stop/0,
+    update_routes/1,
+    insert_route/1,
+    lookup_devaddr/1,
+    lookup_eui/2
+]).
+
+-define(DEVADDRS_ETS, hpr_config_routes_by_devaddr).
+-define(EUIS_ETS, hpr_config_routes_by_eui).
+
+-spec init() -> ok.
+init() ->
+    ?DEVADDRS_ETS = ets:new(?DEVADDRS_ETS, [public, named_table, bag, {read_concurrency, true}]),
+    ?EUIS_ETS = ets:new(?EUIS_ETS, [public, named_table, bag, {read_concurrency, true}]),
+    ok.
+
+-spec stop() -> ok.
+stop() ->
+    true = ets:delete(?DEVADDRS_ETS),
+    true = ets:delete(?EUIS_ETS),
+    ok.
+
+-spec update_routes(map()) -> ok.
+update_routes(#{routes := Routes}) ->
+    {NewDevaddrRows, NewEUIRows} = lists:foldl(
+        fun(RouteConfig, {DevaddrRowsAcc, EUIRowsAcc}) ->
+            Route = hpr_route:new(RouteConfig),
+            DevaddrRowsAcc2 = DevaddrRowsAcc ++ route_to_devaddr_rows(Route),
+            EUIRowsAcc2 = EUIRowsAcc ++ route_to_eui_rows(Route),
+            {DevaddrRowsAcc2, EUIRowsAcc2}
+        end,
+        {[], []},
+        Routes
+    ),
+    true = ets:delete_all_objects(?DEVADDRS_ETS),
+    true = ets:delete_all_objects(?EUIS_ETS),
+    true = ets:insert(?DEVADDRS_ETS, NewDevaddrRows),
+    true = ets:insert(?EUIS_ETS, NewEUIRows),
+    ok.
+
+-spec insert_route(Route :: hpr_route:route()) -> ok.
+insert_route(Route) ->
+    DevaddrRecords = route_to_devaddr_rows(Route),
+    EUIRecords = route_to_eui_rows(Route),
+    true = ets:insert(?DEVADDRS_ETS, DevaddrRecords),
+    true = ets:insert(?EUIS_ETS, EUIRecords),
+    ok.
+
+-spec lookup_devaddr(Devaddr :: non_neg_integer()) -> list(hpr_route:route()).
+lookup_devaddr(Devaddr) ->
+    {ok, NetID} = lora_subnet:parse_netid(Devaddr, big),
+    MS = [
+        {
+            {{'$1', '$2', '$3'}, '$4'},
```

Comment:
This could be made more efficient by moving `NetID` into the pattern match, e.g.,:
```
{   
    {{NetID, '$2', '$3'}, '$4'},
    [   
        {'andalso', {'=<', '$2', Devaddr}, {'=<', Devaddr, '$3'}}
    ],  
    ['$4']
}   
```